### PR TITLE
Add comprehensive unit tests for critical untested code paths

### PR DESCRIPTION
**Pull request by alastairlundy**
## What changed

Added 10 new TUnit unit-test files under `tests/CliInvoke.Tests/` covering the highest-risk previously-untested code paths. No production code was modified.

- `Primitives/UserCredentialTests.cs` - equality operators, GetHashCode consistency, Null sentinel, Dispose (auth primitive)
- `Configuration/UserCredentialSpecTests.cs` - fluent setters, Build(), argument validation, SecureString copy/disposal (auth spec)
- `Internal/ArgumentEscaperTests.cs` - EscapeInner and NeedsQuoting quoting/escaping engine (security-critical)
- `Configuration/ArgumentsSpecTests.cs` - Add/escape/validate/clear/Build (data mutation)
- `Configuration/EnvironmentVariablesSpecTests.cs` - set/duplicate handling/custom comparer/clear/Build
- `Configuration/ProcessResourcePolicySpecTests.cs` - fluent setters, range validation, Build
- `Validation/ProcessResultValidatorTests.cs` - pass/fail, GetValidationFailures, constructors, null guards
- `Validation/CommonValidationRulesTests.cs` - exit-code/regex/standard-error rules
- `Core/Extensions/ProcessResultHelperExtensionsTests.cs` - ThrowIfUnsuccessful, GetFirstOutputLine, GetOutputLines, HasErrors
- `Core/Extensions/ProcessExitConfigurationCreationExtensionsTests.cs` - Default/Create/Graceful/Forceful

## Why it was changed

These are the most critical business-logic, auth, and data-mutation paths that lacked any test coverage. The argument escaper in particular is security-sensitive (shell-injection surface), and the credential/result-validation paths are central to safe process invocation. The new tests document the real, sometimes non-obvious, behavior of these types.

## Testing

- [x] `dotnet test` passes locally on the supported TFMs
- [x] New or updated tests are included for the change
- [x] Behavior-affecting changes are covered by tests

Full suite: 318 tests, 0 failures, run from `tests/CliInvoke.Tests/` on `net10.0`. Note: several tests were written to assert real behavior discovered while implementing, including:
- `UserCredential` compares `SecureString` by reference equality (not value), and `GetHashCode` is only stable for the same `SecureString` instance.
- `UserCredentialSpec` makes its own copy of a set password and disposes only that copy; a `Build()` after `Dispose()` throws `ObjectDisposedException`.
- `ProcessResultHelperExtensions.GetOutputLines` returns a `(string[], string[])` tuple split on `Environment.NewLine` (so it yields a trailing empty entry on Windows).
- `ProcessExitConfiguration.CreateForceful` applies `ForcefulExit` to the timeout policy while cancellation stays `GracefulExit`.

## Documentation

- [ ] README, docs/, or XML doc comments updated (if applicable)
- [ ] VERSION_HISTORY.md updated (if user-visible)
- [x] No docs change needed

## Contribution Policy compliance

- [x] I have read and followed [CONTRIBUTING.md](https://github.com/alastairlundy/CliInvoke/blob/main/CONTRIBUTING.md)
- [x] My branch is up to date with `main` and the diff is focused
- [x] Commit messages are clear and describe the change
- [x] I have not introduced secrets, credentials, or copyrighted content
- [x] For changes that affect resource-owning types, I followed the disposal conventions in the README

## Authorship

- [ ] This PR was written entirely by a human (no AI assistance)
- [x] This PR was Co-Authored by AI (the human author reviewed, edited, and takes responsibility for the final code; commits include the appropriate `Co-authored-by:` trailer(s))

If this PR was Co-Authored by AI, please also fill in the following:

- **AI agent used:** GitHub Copilot CLI
- **AI model used:** hy3

## Summary by CodeRabbit

* **Tests**
  * Expanded automated coverage for command-line argument construction, escaping, and validation.
  * Added coverage for environment variables, credentials, process resource policies, and process-exit behavior.
  * Added tests for process results, output handling, error detection, validation rules, and failure reporting.
  * Added verification of credential equality, disposal behavior, configuration copying, duplicate handling, and edge cases.

### CODE REVIEW

File: tests/CliInvoke.Tests/Configuration/EnvironmentVariablesSpecTests.cs [hunk 1]
```diff
@@ -0,0 +1,203 @@
+/*
+    CliInvoke.Tests
+    Copyright (C) 2024-2026  Alastair Lundy
+
+    This Source Code Form is the terms of the Mozilla Public
```
[the remark this answers is not in the document]
Comment (alastairlundy) [reply tests/CliInvoke.Tests/Configuration/EnvironmentVariablesSpecTests.cs]: Good catch, applied the fix. Restored the omitted "subject to" in the MPL license header.

File: tests/CliInvoke.Tests/Configuration/UserCredentialSpecTests.cs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        await Assert.That(() => spec.Build()).Throws<ObjectDisposedException>();
+        // The caller-owned SecureString is NOT disposed by the spec.
+        await Assert.That(password.Length).IsEqualTo(6);
+    }
+
+    [Test]
+    public async Task Dispose_CanBeCalledMultipleTimes()
+    {
+        // Arrange
+        UserCredentialSpec spec = new UserCredentialSpec().SetPassword(MakeSecureString("secret"));
```
[the remark this answers is not in the document]
Comment (alastairlundy) [reply tests/CliInvoke.Tests/Configuration/UserCredentialSpecTests.cs]: Applied. The caller-owned SecureString is now disposed via a `using` local in `Dispose_CanBeCalledMultipleTimes`.

File: tests/CliInvoke.Tests/Core/Extensions/ProcessResultHelperExtensionsTests.cs [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            result.ThrowIfUnsuccessful(AlwaysFail(), config);
+        }
+        catch (ProcessNotSuccessfulException<ProcessResult> ex)
+        {
+            caught = ex;
+        }
+
+        // Assert
+        await Assert.That(caught).IsNotNull();
+        await Assert.That(caught!.ExecutedProcessInfo).IsNotNull();
```
[the remark this answers is not in the document]
Comment (alastairlundy) [reply tests/CliInvoke.Tests/Core/Extensions/ProcessResultHelperExtensionsTests.cs:75]: Applied. Added an assertion that `caught!.ExecutedProcessInfo.Configuration` is the same reference as the supplied `config`.

[the remark this answers is not in the document]
Comment (alastairlundy) [reply tests/CliInvoke.Tests/Core/Extensions/ProcessResultHelperExtensionsTests.cs:75]: Applied. Added an assertion that `caught!.ExecutedProcessInfo.Configuration` is the same reference as the supplied `config`, verifying the configuration was actually stored.

File: tests/CliInvoke.Tests/Primitives/UserCredentialTests.cs [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+
+        // Assert - the credential holds the same SecureString reference (no copy is made)
+        await Assert.That(credential.Password).IsSameReferenceAs(password);
+
+        // Act - Dispose is idempotent and does not throw
+        credential.Dispose();
+        credential.Dispose();
+
+        // Assert - the credential still exposes the (now disposed) reference
+        await Assert.That(credential.Password).IsSameReferenceAs(password);
```
[the remark this answers is not in the document]
Comment (alastairlundy) [reply tests/CliInvoke.Tests/Primitives/UserCredentialTests.cs:221]: Declining this one. The `UserCredential` constructor stores the `password` reference directly (no copy is made), so `credential.Dispose()` does dispose the caller-owned `SecureString`. The test intentionally verifies that the credential does not make its own copy, and the `using` ensures cleanup at method exit. Asserting `password.Length` throws would test .NET runtime behavior, not the credential's semantics.

File: tests/CliInvoke.Tests/Validation/CommonValidationRulesTests.cs [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+            CommonValidationRules<BufferedProcessResult>.RequiresStandardOutputMatches("goodbye");
+        await Assert.That(rule(MakeBuffered(0, "say hello world", ""))).IsFalse();
+    }
+
+    [Test]
+    public async Task RequiresStandardOutputMatches_NonBufferedResult_Fails()
+    {
+        Func<BufferedProcessResult, bool> rule =
+            CommonValidationRules<BufferedProcessResult>.RequiresStandardOutputMatches("hello");
+        await Assert.That(rule(MakeResult(0) as BufferedProcessResult)).IsFalse();
```
[the remark this answers is not in the document]
Comment (alastairlundy) [reply tests/CliInvoke.Tests/Validation/CommonValidationRulesTests.cs]: Applied. Renamed the tests to `RequiresStandardOutputMatches_NullResult_Fails` and `RequiresStandardErrorIsEmpty_NullResult_Passes`, and passed `null!` directly since `MakeResult(0) as BufferedProcessResult` evaluates to `null` anyway.
